scripts/build_tyndp_h2_demand.py

```diff
@@ -0,0 +1,295 @@
+
+    df_lower_aligned, df_upper_aligned = df_lower.align(
+        df_upper, join="outer", axis=1, fill_value=0
+    )
+
+    missing_in_lower = df_upper.columns.difference(df_lower.columns)
+    missing_in_upper = df_lower.columns.difference(df_upper.columns)
+
+    if len(missing_in_lower) > 0 or len(missing_in_upper) > 0:
+        logger.warning(
+            f"Column mismatch between {year_lower} and {year_upper}. "
+            f"Missing columns filled with zeros. "
+            f"Missing in {year_lower}: {list(missing_in_lower)}, "
+            f"Missing in {year_upper}: {list(missing_in_upper)}"
+        )
+
+    weight = (pyear - year_lower) / (year_upper - year_lower)
+    # Perform linear interpolation
+    result = df_lower_aligned * (1 - weight) + df_upper_aligned * weight
+
+    return result
+
+
+def load_h2_demand(fn: str, scenario: str, pyear: int, cyear: int) -> pd.DataFrame:
+    """Load hydrogen demand files into dictionary of dataframes. Filter for specific climatic year and format data."""
+
+    available_years = get_available_years(fn, scenario)
+    logger.info(
+        f"Scenario {scenario}: Available years: {available_years}, Target year: {pyear}"
+    )
+
+    # If target year exists in data, load it directly
+    if pyear in available_years:
+        logger.info(f"Year {pyear} found in available data. Loading directly.")
+        return load_single_year(fn, scenario, pyear, cyear)
+
+    # Target year not available, do linear interpolation
+    return interpolate_demand(available_years, pyear, fn, scenario, cyear)
+
+
+def check_cyear(cyear: int, scenario: str) -> int:
+    """Check if the climatic year is valid for the given scenario."""
+
+    valid_years = {
+        "NT": np.arange(1983, 2018).tolist(),
+        "DE": [1995, 2008, 2009],
+        "GA": [1995, 2008, 2009],
+    }
+
+    if cyear not in valid_years[scenario]:
+        logger.warning(
+            "Snapshot year doesn't match available TYNDP data. Falling back to 2009."
+        )
+        cyear = 2009
+
+    return cyear
+
+
+if __name__ == "__main__":
+    if "snakemake" not in globals():
+        from _helpers import mock_snakemake
+
+        snakemake = mock_snakemake(
+            "build_tyndp_h2_demand",
+            planning_horizons="2040",
+            clusters="all",
+            configfiles="config/test/config.tyndp.yaml",
+        )
+
+    configure_logging(snakemake)
+    set_scenario_config(snakemake)
+
+    # Parameters
+    scenario = snakemake.params["scenario"]
+    pyear = int(snakemake.wildcards.planning_horizons)
+    snapshots = get_snapshots(snakemake.params.snapshots)
+    cyear = snapshots[0].year
+    fn = snakemake.input.h2_demand
+
+    # Check if climatic year is valid for scenario
+    cyear = check_cyear(cyear, scenario)
+
+    # Load demand with interpolation
+    logger.info(
+        f"Processing H2 demand for scenario: {scenario}, "
+        f"target years: {pyear}, weather year: {cyear}"
```

Comment:
I think we are calling it climate year so far. So probably better to stay consistent with that.
```suggestion
        f"target year: {pyear}, climate year: {cyear}"
```